pilot/pkg/config/clusterregistry/secretcontroller.go

```diff
@@ -199,12 +226,32 @@ func addMemberCluster(s *corev1.Secret, cs *ClusterStore) {
 				Namespace: s.ObjectMeta.Namespace,
 			},
 		}
-		cs.clientConfigs[s.ObjectMeta.Name] = *clientConfig
-		cs.clusters = append(cs.clusters, &cluster)
+		c.cs.clientConfigs[s.ObjectMeta.Name] = *clientConfig
+		c.cs.clusters = append(c.cs.clusters, &cluster)
+		_, client, _ := kube.CreateInterfaceFromClusterConfig(clientConfig)
+		kubectl := kube.NewController(client, kube.ControllerOptions{
+			WatchedNamespace: c.watchedNamespace,
+			ResyncPeriod:     c.resyncInterval,
+			DomainSuffix:     c.domainSufix,
+		})
+
+		c.serviceController.AddRegistry(
+			aggregate.Registry{
+				Name:             serviceregistry.KubernetesRegistry,
+				ClusterID:        GetClusterID(&cluster),
+				ServiceDiscovery: kubectl,
+				ServiceAccounts:  kubectl,
+				Controller:       kubectl,
+			})
+		stopCh := make(chan struct{})
+		_ = kubectl.AppendServiceHandler(func(*model.Service, model.Event) { c.discoveryService.ClearCache() })
+		_ = kubectl.AppendInstanceHandler(func(*model.ServiceInstance, model.Event) { c.discoveryService.ClearCache() })
+
+		go kubectl.Run(stopCh)
 	}
 	// TODO Add exporting a number of cluster to Prometheus
 	// for now for debbuging purposes, print it to the log.
-	log.Infof("Number of clusters in the cluster store: %d", len(cs.clientConfigs))
+	log.Infof("Number of clusters in the cluster store: %d", len(c.cs.clientConfigs))
 }
 
 func deleteMemberCluster(s string, cs *ClusterStore) {
```

Comment:
I think we should also update `deleteMemberCluster` to make sure the service discovery, instance handler etc are also deleted, otherwise, I was still able to visit service cross cluster even if the secret was deleted.

But I think it is ok to handle this in another patch.

Reply:
@gyliu513 Will add `delete` in the following PR